qa/rpc-tests/sc_big_commitment_tree.py

```diff
@@ -0,0 +1,254 @@
+        mark_logs(f"    ScTxsCommitmentTree root: {block_template['scTxsCommitment']}", self.nodes, DEBUG_MODE, 'e')
+
+        # The commitment tree should be the same as in the previous step because we rejected the last TX.
+        assert_equal(current_sc_txs_commitment_tree_root, block_template['scTxsCommitment'])
+        # Also, exceeding_tx should not have been included in the block
+        assert_false(exceeding_tx in [tx['hash'] for tx in block_template['transactions']], "The last TX should not be included in the block candidate!!!")
+
+
+        # Try again, but this time we only send the 4095th FT, so that it will be included in the block candidate
+        # We use createrawtransaction to avoid dependencies to the previous tx
+        mark_logs(f"Adding 1 FT for sc {sc_id} to the mempool", self.nodes, DEBUG_MODE, 'g')
+        try:
+            tx_data = {"address": "aabb", "amount": 0.003, "scid": sc_id, "mcReturnAddress": node1_address}
+            rawtx = self.nodes[0].createrawtransaction([],{},[], [], [tx_data])
+            fundedtx = self.nodes[0].fundrawtransaction(rawtx)
+            sigRawtx = self.nodes[0].signrawtransaction(fundedtx['hex'])
+            last_tx = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+        except JSONRPCException as e:
+            print(e.error['message'])
+            assert(False)
+        mark_logs(f"Sent transaction FT #4095", self.nodes, DEBUG_MODE, 'c')
+        mark_logs(f"  5 - {last_tx}", self.nodes, DEBUG_MODE, 'e')
+        self.sync_all()
+
+        # Ask for a new block template
+        block_template = self.wait_for_block_template(0, i + 2)
+        assert(block_template is not None)
+        current_sc_txs_commitment_tree_root = block_template['scTxsCommitment']
+        # Check it has been included in the block candidate
+        assert_true(last_tx in [tx['hash'] for tx in block_template['transactions']], "The last TX should not be included in the block candidate!!!")
+        mark_logs(f"getblocktemplate() ->", self.nodes, DEBUG_MODE, 'e')
+        for k, transaction in enumerate(block_template['transactions']):
+            mark_logs(f"  {k} - {transaction['hash']}", self.nodes, DEBUG_MODE, 'e')
+        mark_logs(f"    ScTxsCommitmentTree root: {block_template['scTxsCommitment']}", self.nodes, DEBUG_MODE, 'e')
+
+
+        # Try again adding an exceeding tx to the block candidate
+        mark_logs(f"Adding 1 FT for sc {sc_id} to the mempool", self.nodes, DEBUG_MODE, 'y')
+        try:
+            exceeding_tx2 = self.nodes[1].sc_send(forwardTransferOuts, {"fee": Decimal(0.0001 * 7), "minconf": 0})
+        except JSONRPCException as e:
+            print(e.error['message'])
+            assert(False)
+        mark_logs(f"Sent transaction FT #4096", self.nodes, DEBUG_MODE, 'c')
+        mark_logs(f"  6 - {exceeding_tx2}", self.nodes, DEBUG_MODE, 'e')
+        self.sync_all()
+
+        # Ask for a new block template
+        time.sleep(5)
```

Comment:
Can we get rid of this sleep now that we have the wait_for_block_template() function?

```suggestion
```

Reply:
Addressed in 9ce2a8aff3